aatools/food_insecurity/fewsnet.py

```diff
@@ -0,0 +1,218 @@
+        only the year and month part are used
+        this commonly refers to the month of the Current Situation period
+    region_name : str
+        name of the region to which the `iso2` belongs,
+        e.g. "east-africa"
+    region_code : str
+        code that refers to the `region_name,
+        e.g. "EA"
+    output_dir : Path
+        path to dir to which the data should be written
+    use_cache : bool
+        if True, don't download if output_dir already exists
+
+    Returns
+    -------
+    region_data : bool
+        if True, region data for the given date and region name exists
+    """
+    base_url_region = (
+        "https://fews.net/data_portal_download/download?"
+        "data_file_path=http://shapefiles.fews.net.s3.amazonaws.com/HFIC/"
+    )
+    url_region = (
+        f"{base_url_region}{region_code}/"
+        f"{region_name}{date.strftime('%Y%m')}.zip"
+    )
+    zip_path_region = output_dir / f"{region_code}{date.strftime('%Y%m')}.zip"
+    output_dir_region = output_dir / f"{region_code}{date.strftime('%Y%m')}"
+
+    if not output_dir_region.exists() or use_cache is False:
+        region_data = download_zip(
+            url_region, zip_path_region, output_dir_region
+        )
+    else:
+        region_data = True
+    return region_data
+
+
+def download_fewsnet(
+    date: datetime,
+    iso2: str,
+    region_name: str,
+    region_code: str,
+    # question: is it good practice to have
+    # output_dir as input arg or define that with a config?
+    # question: is it too much to require it to be a Path object?
+    output_dir: Path,
+    use_cache=True,
+):
+    """
+    Retrieve the raw fewsnet data.
+
+    Depending on the region and date, this data is published per region or
+    per country. This function retrieves the country data
+    if it exists, and else the regional data for `date` and `iso2`.
+
+    Parameters
+    ----------
+    date : datetime
+        date for which the data should be downloaded
+        only the year and month part are used
+        this commonly refers to the month of the Current Situation period
+    iso2 : str
+        iso2 code of the country of interest
+    region_name : str
+        name of the region to which the `iso2` belongs,
+        e.g. "east-africa"
+    region_code : str
+        code that refers to the `region_name,
+        e.g. "EA"
+    output_dir : Path
+        path to dir to which the data should be written
+    use_cache : bool
+        if True, don't download if output_dir already exists
+    """
+    # question: is there a method to force the input to be upper-cased?
+    # upper case iso2 and region_code since they are upper-cased
+    # in fewsnet's url
+    iso2 = iso2.upper()
+    region_code = region_code.upper()
+
+    # we prefer the country data as this more nicely structured
+    # thus first check if that is available
+    # question: is it a good or bad idea to split this to two functions?
```

Comment:
Definitely good to split this up! Also, I would make the functions private (add an underscore to the front of their names) if they are only meant to be used within this module